session/pingpong/consumer_balance_tracker.go

```diff
@@ -140,9 +140,27 @@ func (cbt *ConsumerBalanceTracker) Subscribe(bus eventbus.Subscriber) error {
 	if err != nil {
 		return err
 	}
+	err = bus.SubscribeAsync(event.AppTopicSettlementComplete, cbt.handleSettlementComplete)
+	if err != nil {
+		return err
+	}
+	err = bus.SubscribeAsync(event.AppTopicWithdrawalRequested, cbt.handleWithdrawalRequested)
+	if err != nil {
+		return err
+	}
 	return bus.SubscribeAsync(identity.AppTopicIdentityUnlock, cbt.handleUnlockEvent)
 }
 
+// settlements increase balance on the chain they are settled on.
+func (cbt *ConsumerBalanceTracker) handleSettlementComplete(ev event.AppEventSettlementComplete) {
+	go cbt.aggressiveSync(ev.ChainID, ev.ProviderID, cbt.cfg.FastSync.Timeout, cbt.cfg.FastSync.Interval)
```

Comment:
At this point it seems like it'd be really to not be able to hit `cbt.aggressiveSync` or `cbt.periodicSync` multiple times. As in if you called it once and it's running - you can't call it again (or calls are ignored) until it finishes. Lots of different places call it and it will start to overlap which will cost us requests for nothing in the future or if something goes wrong 🤷🏻

Reply:
k, will fix

Reply:
Added a balance syncer, that will only re-launch the sync job in case the params match.